src/utils/object.ts

```diff
@@ -216,6 +216,56 @@ export function camelizeObjectKeys(obj: { [key: string]: any }) {
   return newObj;
 }
 
+/**
+ * camelizeKeys recursively turns object keys camelCased, works on
+ *   arrays of objects, and object-key's array values, which may
+ *   contain column names
+ * This does not replace camelizedObjectKeys because of its
+ *   aggressive (recursive) nature, which is a little more greedy
+ *
+ * @param arg <Array|object|string>
+ * @returns An object with camelized keys or camelized string
+ */
+export function camelizeKeys(arg: any) {
+  let response: any;
+
+  if (typeof arg === 'string') {
+    return camelize(arg);
+  } else if (Array.isArray(arg)) {
+    response = [];
+
+    for (const [i, element] of arg.entries()) {
+      if (typeof element === 'string') {
+        response[i] = camelize(element);
+      } else { // array/object/etc
+        response[i] = camelizeKeys(element);
+      }
+    }
+  } else if (typeof arg === 'object') {
+    response = {};
+
+    for (const key of Object.keys(arg)) {
+      const value = arg[key];
+
+      if (typeof key === 'string') {
+        // TODO: consider camelizizing values that are columns and reference columns:
+        //    let isColumnKey = key.toLowerCase().includes('column')
+        //    response[camelize(key)] = isColumnKey ? camelizeKeys(value) : value
+        //    REM: camelizing values may have considerable impact on user expectations
+        //      this requires a lot of consideration
+        response[camelize(key)] = value;
```

Comment:
`value` should be passed to `camelizeKeys` if it's a plain object or an array

Reply:
Are you sure that's what you'd want?  That would change `fk_id` to `fkId`.  Isn't the raw form desired?

Reply:
I don't understand the purpose of this method then. If it's not recursively camelizing keys, then the other method is sufficient

Reply:
I think this would benefit from some test cases to demonstrate what's expected

I also might walk through the developmental process:
1. This originated after working on Foreign Keys and more specifically, composite foreign keys, which required moving from string values to array values
2. I was going to camelize everything -- keys and values -- which explains the recursive nature.  Consider the following example of data I think I came across:
   ```javascript
   [ { books_table: { column_names: ['fk_col1', 'fk_col2'] }, authors_table: { ... } ]
   ``` 

   I initially attempted to use `camelizeObjectKeys`, but here is what the results would be for above object:

   function | outcome
   --- | ---
   foreignKeys.map(camelizeObjectKeys) | `[ { booksTable: { column_names: ['fk_col1', 'fk_col2'] }, { authorsTable: { ... } }]` <br> The existing function would only update the first set of keys
   camelizeKeys(foreignKeys) <br>- or -<br> foreignKeys.map(foreignKeys) | `[ { booksTable: { columnNames: ['fkCol1', 'fkCol2'] }, { authorsTable: { ... } }]` <br> Iterates over everything.  Keys and values.
3. Then I thought it was a bad idea to convert the values, given that a user (or internal methods) may want to use the raw value. At this point I considered scrapping the entire function.
4. Then I came across dialects or differences between the format of getForeignKeysForTable vs getForeignKeysForTables vs getForeignKeyReferences, ..., and the results varied.  I think between a simple object `{constraintName: 'foo'}`, keyed objects `{ foreign_key: { tableName: 'users', schema: 'public'} }` , object that had table name as key and string values in an array (`constraints = { read_only_users: ['fk_1', 'fk_2', 'ref_3'] }`.  Sometimes the array of objects had a key (`[{ foo: { ... }, {bar: {...}}]`) or nested object-value, sometimes they didn't (`[ {...}, {...}]`).
   
   Instead of doing type/format detection, what I was really after was to camelizethe keys of all objects, which could be 2-3 levels deep and may or may not be in an array.  That's the landed code with results below:
   command | explanation
    --- | ---
   camelizeKeys(foreignKeys) <br>- or -<br> foreignKeys.map(foreignKeys) | iterates over everything but object-values <br> `[ { booksTable: { columnNames: ['fk_col1', 'fkcol2'] }, { authorsTable: { ... } }]`
   
   Versatile enough to handle arrays, arrays-of-objects, objects with object-values (e.g., `{ references: { table: {table_name: 'table', schema: 'public'} }`), but I consider this will be more useful in the time to come as I think the eventual state would be to have more layered data, which the API providing methods (or options) that might simplify the results if desired.  
   
---

**More**: I don't have a good understanding of the Sequelize ORM as a user to know how it behaves. In some ORMs you can use values to perform linkages, or the ORM might do a reverse translation.  For instance, they might do something like `orm.findTables({ where: fkId: 'parts_id' })` to retrieve tables where the foreign key is parts_id.  Notice that on the table the foreign key might be named `fk_id`, but the ORM would accept the camelized user input and convert it back to `fk_id`.  I don't know if Sequelize accomplishes those reverse-translation for column names.

---

Note: My last comment above was in response to your comment, which is suggesting a change to the code above (I confused myself when I looked at the code and saw my comment)

Reply:
I see a first flaw then, if `books_table` is the name of the table, then it should not be camelized. It should be preserved as-is.

However, the `column_names` key (assuming it's a constant and not an actual table name) should be camelized.

> Notice that on the table the foreign key might be named `fk_id`, but the ORM would accept the camelized user input and convert it back to `fk_id`

We do support that, but that's a separate concern which does not impact this. This needs to return table names & column names as-is from the database. The Model will handle any necessary mapping separately.

Reply:
This is helpful! Tell me if I've got this wrong, but what I'm hearing is:
- the object names (e.g., `book_authors` table) should never be camelized

If that's the case, then yes, I absolutely need to review this and see where it'd be used. Oddly, I only see it in the Postgres QI and I thought I used it in a few places.  I also need to point out this should be a private function.  The intention was to use it primarily around the foreignKey/constraint lookup-queries to simplify the calling code.  

Are the rules/conditions/specs/conventions like this laid out anywhere for v7? That might save us some time in the future.